File: src/reservation/internal/model/reservation.go
package model

type Reservation struct {
	ReservationUID string `json:"reservationUid"`
	Username       string `json:"username"`
	BookUID        string `json:"bookUid"`
	LibraryUID     string `json:"libraryUid"`
	Status         string `json:"status"`
	StartDate      string `json:"startDate"`
	TillDate       string `json:"tillDate"`
}

type ReservationRequest struct {
	BookUID    string `json:"bookUid"`
	LibraryUID string `json:"libraryUid"`
	TillDate   string `json:"tillDate"`
}

File: src/reservation/entity/reservation.py
from datetime import datetime
from dataclasses import dataclass
from enum import Enum

class ReservationStatus(Enum):
    RENTED = "RENTED"
    RETURNED = "RETURNED"
    EXPIRED = "EXPIRED"

@dataclass
class Reservation:
    id: int
    reservation_uid: str
    username: str
    book_uid: str
    library_uid: str
    status: ReservationStatus
    start_date: datetime
    till_date: datetime
